catalyst-gateway/bin/src/service/common/types/cardano/query/stake_or_voter.rs

```diff
@@ -198,3 +203,21 @@ impl TryInto<common::types::generic::ed25519_public_key::Ed25519HexEncodedPublic
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::StakeAddressOrPublicKey;
+
+    #[test]
+    fn hex_to_stake_or_voter() {
```

Comment:
Could also add some comments from where you got these test vectors (maybe a link to cip 18).

Reply:
added[ this](https://cexplorer.io/article/understanding-cardano-addresses) as its the best resource